pkg/ebpf/manager.go

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package ebpf
+
+import (
+	"fmt"
+	"io"
+
+	manager "github.com/DataDog/ebpf-manager"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// Manager wraps ebpf-manager.Manager, adding a property with the list of enabled modifiers
+// for this instance.
+type Manager struct {
+	*manager.Manager
+	EnabledModifiers []string // List of enabled modifiers
+}
+
+// Names of modifiers are defined here to avoid duplication and so that they can be
+// used in the list of default modifiers
+const (
+	// NoModifiers is a constant to be used when creating a new manager to avoid initializing the default modifiers
+	NoModifiers = "NO_MODIFIERS"
+	// PrintkModifier is the name of the modifier that patches the printk function
+	PrintkModifier = "printk"
+)
+
+// defaultModifiers is a list of modifiers that are enabled by default when the callers don't provide
+// a specific list
+var defaultModifiers = []string{PrintkModifier}
+
+// NewManager creates a manager wrapper.
+// If the modifiers list is empty, it will be initialized with the default modifiers.
+// Pass manager.NO_MODIFIERS to avoid initializing the default modifiers.
+func NewManager(mgr *manager.Manager, modifiers ...string) *Manager {
```

Comment:
Do we even need the strings? Could we use types instead and each modifier expose a global `var PrintKPatcher Modifier`, then the call becomes `NewManager(mgr, modifiers.PrintKPatcher)`

Reply:
I think this might remove the need to register modifiers as well. The globals could start out `nil` and then triggered to be created from `ebpf.Setup()` when we know eBPF is required?